monai/networks/utils.py

```diff
@@ -531,11 +531,13 @@ def copy_model_state(
             updated_keys.append(dst_key)
     for s in mapping if mapping else {}:
         dst_key = f"{dst_prefix}{mapping[s]}"
-        if dst_key in dst_dict and dst_key not in to_skip:
-            if dst_dict[dst_key].shape != src_dict[s].shape:
-                warnings.warn(f"Param. shape changed from {dst_dict[dst_key].shape} to {src_dict[s].shape}.")
-            dst_dict[dst_key] = src_dict[s]
-            updated_keys.append(dst_key)
+        src_keys = sorted({s_key for s_key in src_dict if s_key not in to_skip and re.compile(s).search(s_key)})
+        dst_keys = sorted({d_key for d_key in dst_dict if d_key not in to_skip and re.compile(dst_key).search(d_key)})
+        for _src_key, _dst_key in zip(src_keys, dst_keys):
```

Comment:
this `zip` assumes 1 to 1 mapping for the first N keys, I can see in practice there could be missing keys in the middle of `dst_keys` or `src_keys` and will cause mismatches... in my opinion defining this kind of search pattern is complicated, but if there's a use case I'm fine with this new feature.

Reply:
Thanks, Wenqi, I know your concern. I didn't find a good way to do such a thing when we have some complex rules to match weights such as the https://github.com/Project-MONAI/MONAI/issues/6552#issuecomment-1587582734,  
Maybe we should leave these filter logic to users by just exposing a `filter` arg.
Or maybe I can add some warning here when the length of `dst_keys` and `src_keys` mismatch.

Anyway, let's wait and see what others say.